### PR TITLE
Add `PointwiseFeedForward`

### PR DESCRIPTION
**Pull request by danieldk**

### CODE REVIEW

File: src/layers/feedforward.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+use candle_core::{Module, Tensor};
+use candle_nn::{linear, linear_no_bias, Linear, VarBuilder};
+
+use crate::error::Result;
+
+/// Feed-forward layers.
+
+/// Point-wise feed-forward layer (_Vaswani et al., 2017_).
```
Comment (shadeMe) [inline src/layers/feedforward.rs:8]: Does `rustdoc` support this syntax? Doesn't look like regular markdown to me.

File: src/layers/feedforward.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    /// * `use_gate` - Use Gated Linear Units.
+    pub fn new(
+        vb: VarBuilder,
+        activation: Box<dyn Module>,
+        hidden_width: usize,
+        intermediate_width: usize,
+        use_bias: bool,
+        use_gate: bool,
+    ) -> Result<Self> {
+        let linear_cons = if use_bias { linear } else { linear_no_bias };
```
Comment (shadeMe) [inline src/layers/feedforward.rs]: Minor nitpick: Maybe `linear_ctor` instead of `linear_cons`?

File: src/layers/feedforward.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    pub gate: Option<Linear>,
+    pub intermediate: Linear,
+    pub output: Linear,
+}
+
+impl PointwiseFeedForward {
+    /// Construct a point-wise feed-forward layer.
+    ///
+    /// * `vb` - Variable store.
+    /// * `activation` - non-linearity.
```
Comment (shadeMe) [inline src/layers/feedforward.rs]:
```suggestion
    /// * `activation` - Non-linearity.
```

File: src/layers/feedforward.rs (hunk 1 above)
Comment (danieldk) [reply src/layers/feedforward.rs:8]: It does, makes the _Vaswani et al., 2017_ part italic.

File: src/layers/feedforward.rs (hunk 2 above)
Comment (danieldk) [reply src/layers/feedforward.rs]: My FP leanings 😁. Fixed.

File: src/layers/feedforward.rs (hunk 3 above)
Comment (danieldk) [reply src/layers/feedforward.rs]: Fixed.

**Pull request merged**